room_assistance/indico_room_assistance/migrations/20190220_1138_b160812c186b_add_startup_assistance_requests_table.py

```diff
@@ -0,0 +1,90 @@
+"""Add room assistance requests table
+
+Revision ID: b160812c186b
+Revises:
+Create Date: 2019-02-20 11:38:08.691528
+"""
+
+from __future__ import unicode_literals
+
+from alembic import context, op
+from terminaltables import AsciiTable
+
+
+# revision identifiers, used by Alembic.
+revision = 'b160812c186b'
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    if context.is_offline_mode():
+        raise Exception('This upgrade is only possible in online mode')
+
+    conn = op.get_bind()
+    conn.execute('''
+        INSERT INTO events.requests (event_id, type, state, data, created_by_id, created_dt)
+        SELECT
+            rvl.linked_event_id,
+            'room-assistance',
+            1,
+            '{}',
+            (SELECT id FROM users.users WHERE is_system),
+            rv.start_dt
+        FROM roombooking.reservations rv
+        JOIN roombooking.reservation_links rvl ON rv.link_id = rvl.id AND rvl.link_type = 2
+        WHERE rv.needs_assistance
+    ''')
+
+    invalid_reservations = conn.execute('''
+        SELECT rv.id, rv.room_id, rvl.id, rv.start_dt, rv.end_dt, rv.booked_for_name
+        FROM roombooking.reservations rv
+        LEFT JOIN roombooking.reservation_links rvl ON rv.link_id = rvl.id
+        WHERE rv.needs_assistance
+        AND (rv.link_id IS NULL OR rvl.link_type != 2)
+        AND rv.start_dt > NOW()
+    ''')
+
+    table_data = [['ID', 'Room ID', 'Link ID', 'Start date', 'End date', 'Booked for']]
```

Comment:
Link ID is always None, because your never have a reservation link when migrating the data

Reply:
What about printing the room name instead of ID? If we sent the current list to the assistance guys they'd probably hate us

Reply:
Are you sure LINK ID is always None? We have a migration that migrates that data in the core.

Reply:
`AND (rv.link_id IS NULL OR rvl.link_type != 2)` - I guess 2 is event, and we only migrate event links since any other links did not exist before